internal/controllers/reference/reference.go

```diff
@@ -57,6 +58,45 @@ func UpdateReferencesToSecret(
 	return removeOutdatedReferencesToSecret(ctx, indexers, c, dataplaneClient, referrer, referencedSecretNameMap)
 }
 
+// UpdateReferencesToConfigMap updates the reference records between referrer and each ConfigMap
+// in namespacedNames in record cache.
+func UpdateReferencesToConfigMap(
```

Comment:
This also looks eerily similar to `UpdateReferencesToSecret`. Is there any chance to deduplicate this and share some common parts?

Reply:
Done in de81f12fd9a7aa3bf513b2e2db26b2d7de826eec